variables-workers.tf

```diff
@@ -0,0 +1,232 @@
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+#
+# Cluster
+#
+
+variable "cluster_id" {
+  default     = null
+  description = "An existing OKE cluster OCID when `create_cluster = false`."
+  type        = string
+}
+
+variable "cluster_ca_cert" {
+  default     = null
+  description = "Base64+PEM-encoded cluster CA certificate for unmanaged instance pools. Determined automatically when 'create_cluster' = true or 'cluster_id' is provided."
+  type        = string
+}
+
+variable "cluster_dns" {
+  default     = "10.96.5.5"
+  description = "Cluster DNS resolver IP address. The provided value used with default 'servicesCidr' 10.96.0.0/16 should only require modification in the case of conflict."
+  type        = string
+}
+
+#
+# Worker pools
+#
+
+variable "worker_pools" {
+  default     = {}
+  description = "Tuple of OKE worker pools where each key maps to the OCID of an OCI resource, and value contains its definition."
+  type        = any
+}
+
+variable "worker_pool_enabled" {
+  default     = true
+  description = "Whether to apply resources for a group when unspecified."
+  type        = bool
+}
+
+variable "worker_pool_mode" {
+  default     = "node-pool"
+  description = "Default management mode for worker pools when unspecified. Only 'node-pool' is currently supported."
+  type        = string
+  validation {
+    condition     = contains(["node-pool", "instance-pool", "cluster-network"], var.worker_pool_mode)
+    error_message = "Accepted values are node-pool, instance-pool, or cluster-network"
+  }
+}
+
+variable "worker_pool_size" {
+  default     = 0
+  description = "Default size for worker pools when unspecified."
+  type        = number
+}
+
+variable "output_worker_detail" {
+  default     = false
+  description = "Whether to include detailed output for worker pool configuration."
+  type        = bool
+}
+
+#
+# Workers: network
+#
+
+variable "worker_type" {
+  default     = "private"
+  description = "Whether to provision public or private workers."
+  type        = string
+  validation {
+    condition     = contains(["public", "private"], var.worker_type)
+    error_message = "Accepted values are public or private."
+  }
+}
+
+variable "worker_nsg_ids" {
+  default     = []
+  description = "An additional list of network security group (NSG) IDs for node security."
+  type        = list(string)
+}
+
+variable "pod_nsg_ids" {
+  default     = []
+  description = "An additional list of network security group (NSG) IDs for pod security."
+  type        = list(string)
+}
+
+variable "kubeproxy_mode" {
+  default     = "iptables"
+  description = "The mode in which to run kube-proxy."
+  type        = string
+
+  validation {
+    condition     = contains(["iptables", "ipvs"], var.kubeproxy_mode)
+    error_message = "Accepted values are iptables or ipvs."
+  }
+}
+
+#
+# Workers: instance
+#
+
+variable "worker_block_volume_type" {
```

Comment:
Can you please also add the ability to configure the vpu?

Reply:
Good idea, but let's do this in another change; there are other storage-related configs that would be worth surfacing as well, but warrant explicit testing and possible preconditions.